src/ad-bidders/a9/index.js

```diff
@@ -140,27 +108,65 @@ export class A9 extends BaseBidder {
 		return definition;
 	}
 
-	getBestPrice(slotName) {
-		return this.priceMap[slotName] ? { a9: this.priceMap[slotName] } : {};
+	getBidTargetingAndKeys(bid) {
+		return this.bidderConfig.dealsEnabled
+			? {
+				keys: bid.helpers.targetingKeys,
+				bidTargeting: bid.targeting
+			}
+			: {
+				keys: this.apstag.targetingKeys(),
+				bidTargeting: bid
+			};
 	}
 
-	getPrices() {
-		return this.priceMap;
+	updateBidSlot(slotName, keys, bidTargeting) {
+		this.bids[slotName] = {};
+		keys.forEach((key) => {
+			if (this.targetingKeys.indexOf(key) === -1) {
+				this.targetingKeys.push(key);
+			}
+			this.bids[slotName][key] = bidTargeting[key];
+		});
 	}
 
-	getTargetingKeysToReset() {
-		return this.targetingKeys;
+	async callBids(onResponse) {
+		if (this.cmp) {
```

Comment:
It's assigned in the constructor, can it ever be false?

Reply:
Tak powinno być:
```suggestion
		if (this.cmp.exists) {
```
Dlatego że `window.__cmp` może być falsy.